geomstats/geometry/discrete_curves.py

```diff
@@ -572,6 +618,47 @@ def dist(self, point_a, point_b, **kwargs):
 
         return dist
 
+    # TODO (alebrigant): generalize from_vector_to_diagonal_matrix to k
+    # diagonals and modify tensorflow vstack and hstack.
+    def space_derivative(self, curve):
+        """Compute space derivative of curve using centered differences.
+
+        Parameters
+        ----------
+        curve : array-like, shape=[..., n_points, ambient_dim]
+            Discrete curve.
+
+        Returns
+        -------
+        space_deriv : array-like, shape=[...,n_points, ambient_dim]
+        """
+        n_points = curve.shape[-2]
+
+        vec_1 = gs.squeeze(gs.hstack((
+            gs.array([[-1.]]),
+            gs.zeros((1, n_points - 2)),
+            gs.array([[1.]]))))
+        vec_2 = gs.squeeze(gs.hstack((
+            1 / 2 * gs.ones((1, n_points - 2)),
+            gs.array([[1.]]))))
+        vec_3 = gs.squeeze(gs.hstack((
+            gs.array([[1.]]),
+            1 / 2 * gs.ones((1, n_points - 2)))))
+
+        mat_1 = from_vector_to_diagonal_matrix(vec_1)
```

Comment:
For `mat_1` you can use `gs.array_from_sparse([(0, 0), (n_points - 1,) * 2 ], [-1, 1], [n_points, n_points])`

Reply:
Thanks I didn't know we had this function. I'm not sure it really improves readability though in this case.